app/models/katello/pool.rb

```diff
@@ -50,13 +50,16 @@ class Pool < Katello::Model
 
     validates_lengths_from_database
 
-    validates :subscription, :presence => true
+    validates :subscription, :presence => { :message => N_("cannot be blank. If this is during a manifest import or refresh, try running foreman-rake katello:reimport.") }
     validates :organization, :presence => true
 
     validate :subscription_matches_organization
 
     def subscription_matches_organization
-      errors.add(:base, _("A Pool and its Subscription cannot belong to different organizations")) unless subscription&.organization_id == self.organization_id
+      return if subscription&.organization_id.blank? || self.organization_id.blank? # let other validations catch this
```

Comment:
I had the same thought as you did when I wrote this validation. But then I thought: when would either of those be blank? The underlying issue you're working on has triggered this validation so I think it's good to leave it alone and in its more strict form even if it's triggered by Some Other Problem

Reply:
But aren't we both a fan of useful, accurate error messages? This was triggered not because `subscription.organization_id` was blank, but because `subscription` was blank. So it's not true that the organizations are different-- the error message is just inaccurate.  In this case it would fall back to the already-existing `validates :subscription, :presence => true` validation-- which is why I added the message to that.

Reply:
basically the flow is
1. you try to import a bad manifest and hit this issue
2. you see the manifest refresh/import fail with "If this is during a manifest import or refresh, try running foreman-rake katello:reimport"
3. you run the rake task and all is fixed!

Reply:
I see your point! +1 to including it.

Reply:
Lightbulb moment? Rather than checking for presence/absence of org and subscription, what about checking if there any _errors_ for those fields? This changes the logic from: 'skip this validation if these fields are blank' to 'skip this validation if we already have errors on subscription and org'

I know this is a minor detail in the bigger picture here, but wanted to share my thought now that I slept on it

Reply:
Sounds like this will work, but only if Rails runs custom validations after stock validations. I'll check on that.